gap/attr.gi

```diff
@@ -8,6 +8,85 @@
 #############################################################################
 ##
 
+# The next method is (yet another) DFS as described in
+# http://www.eecs.wsu.edu/~holder/courses/CptS223/spr08/slides/graphapps.pdf
+
+InstallMethod(ArticulationPoints, "for a digraph", [IsDigraph],
+function(digraph)
+  local copy, nbs, counter, visited, num, low, parent, points, v_stack,
+  w_stack, depth, 1st_root_nb, v, w, i;
+
+  if HasIsConnectedDigraph(digraph) and not IsConnectedDigraph(digraph) then
+    return [];
+  elif DigraphNrVertices(digraph) < 1 then
+    return [];
+  elif not IsSymmetricDigraph(digraph) then
+    copy := DigraphSymmetricClosure(digraph);
+  else
+    copy := digraph;
+  fi;
+  nbs := OutNeighbours(copy);
+
+  counter     := 0;
+  visited     := BlistList([1 .. DigraphNrVertices(copy)], []);
+  num         := [];
+  low         := [];
+  parent      := [1];
+  points      := [];
+
+  v_stack     := [1];
+  w_stack     := [0];
+  depth       := 1;
```

Comment:
Many of these lines, like `points      := [];`, would not pass our own linting checks if they were in the Semigroups package, due to multiple spaces.  I don't mind, but would you rather Digraphs conformed to gaplint standard?